pytorch_lightning/core/lightning.py

```diff
@@ -1610,14 +1610,24 @@ def tbptt_split_batch(self, batch, split_size):
 
         return splits
 
-    def summarize(self, mode: Optional[str] = ModelSummary.MODE_DEFAULT) -> Optional[ModelSummary]:
+    def summarize(self, mode: Optional[str] = "top", max_depth: Optional[int] = None) -> Optional[ModelSummary]:
         model_summary = None
```

Comment:
unrelated to this PR, but I feel `summarize` shouldn't be a method on the core lightning module API. it seems like a utility function that accepts the LightningModule as input along with the mode/depth would be preferable instead

Reply:
Related to your comment @ananthsub

https://github.com/PyTorchLightning/pytorch-lightning/issues/4541